prompt2model/utils/parse_responses.py

```diff
@@ -97,42 +62,46 @@ def parse_json(
     return final_response
 
 
-def parse_reranking_results(
-    response: openai.Completion,
-) -> dict | None:
-    """Parse a formatted string and extract numbers for reranking results.
+def find_rightmost_brackets(text: str) -> str | None:
+    """Find the rightmost complete set of brackets in a string."""
+    stack = []
+    for i, char in enumerate(reversed(text)):
+        if char == "}":
+            stack.append(len(text) - i - 1)
+        elif char == "{" and stack:
+            start = len(text) - i - 1
+            end = stack.pop()
+            if not stack:  # Found the rightmost complete set
+                return text[start : end + 1]
+    return None
+
+
+def parse_dataset_config_responses(response: openai.ChatCompletion) -> dict:
+    """Parses the response to extract relevant information from dataset/configuration.
+
+    LLMs can return the dataset configuration in different formats -
+    usually either between ** ** or as a sentence.
 
     Args:
-        response_string: A string containing three comma separated fields'.
+        response (str): The response containing the dataset configuration.
 
     Returns:
-        a dict with (dataset_name, config_name, confidence_level) fields
-        or None if the response cannot be parsed
-
+        dict: The extracted relevant information from the dataset configuration.
```

Comment:
no need to include a type hint here
```suggestion
        The extracted relevant information from the dataset configuration.
```